src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs

```diff
@@ -34,26 +35,38 @@ public class RunReadyToRunCompiler : ToolTask
         private string _createPDBCommand;
 
         private bool IsPdbCompilation => !String.IsNullOrEmpty(_createPDBCommand);
+        private bool ActuallyUseCrossgen2 => UseCrossgen2 && !IsPdbCompilation;
+
+        private string DotNetHostPath => Crossgen2Tool?.GetMetadata(MetadataKeys.DotNetHostPath);
+
+        private bool Crossgen2IsVersion5
+        {
+            get
+            {
+                string version5 = Crossgen2Tool?.GetMetadata(MetadataKeys.IsVersion5);
+                return version5 != null && bool.Parse(version5);
```

Comment:
`bool.Parse` may throw, e.g., when you pass an empty string.
```suggestion
                return bool.TryParse(version5, out bool value) && value;
```

Reply:
Hmm, wouldn't it be safer to check IsNullOrEmpty and then make bool.Parse throw when garbage is passed in?

Reply:
Added empty string check in 8th commit.

Reply:
In project and targets file we usually use `'$(foo)' == 'true'` and `'$(foo)' != 'true'` comparisons and do not throw on incorrect Boolean values.  Note that in another place you use `TryParse` for the same check:
```C#
bool.TryParse(Crossgen2Tool.GetMetadata(MetadataKeys.IsVersion5), out crossgen2IsVersion5);
```
If you wish to have just the `!IsNullOrEmpty` check, I am fine with that.

Reply:
I have modified the other place in PrepareR2RC to use the same algorithm as Crossgen2IsVersion5 in RunR2RC. I do know that in msbuild scripts we typically check for "true" and I'm well acquainted with all the obscure bugs that can happen once someone accidentally passes an unexpected parameter value.